hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/contract/AbstractContractGetFileStatusTest.java

```diff
@@ -272,38 +275,22 @@ public void testListFilesNoDir() throws Throwable {
     }
   }
 
-  @Test
+  @Test (expected = FileNotFoundException.class)
   public void testLocatedStatusNoDir() throws Throwable {
     describe("test the LocatedStatus call on a path which is not present");
-    try {
-      RemoteIterator<LocatedFileStatus> iterator
-          = getFileSystem().listLocatedStatus(path("missing"));
-      fail("Expected an exception, got an iterator: " + iterator);
-    } catch (FileNotFoundException expected) {
-      // expected
-    }
+    getFileSystem().listLocatedStatus(path("missing"));
```

Comment:
I'd prefer retaining the intercept() code of trunk. I know java 7 hates l-expressions, but we've done the backport elsewhere: see org.apache.hadoop.fs.s3a.s3guard.AbstractS3GuardToolTestBase for some examples. IntellJ IDEA will actually do the conversion from an lambda expression to a Callable for you if you ask it nicely

Reply:
I'd prefer retaining the intercept() code of trunk. I know java 7 hates l-expressions, but we've done the backport elsewhere: see org.apache.hadoop.fs.s3a.s3guard.AbstractS3GuardToolTestBase for some examples. IntellJ IDEA will actually do the conversion from an lambda expression to a Callable for you if you ask it nicely

Reply:
Yes I have fixed it.